src/content/blog/astro-4120.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: "Astro 4.12: Server Islands"
+description: Astro 4.12 is now available! This release includes a first look at Server Islands, improvements to pagination and syntax highlighting, and more.
+publishDate: "July 18, 2024"
+authors:
+  - erika
+  - matthew
+coverImage: "/src/content/blog/_images/astro-4120/header_astro_4_12.webp"
+socialImage: "/src/content/blog/_images/astro-4120/og_image_4_12.webp"
+lang: "en"
+---
+
+import BlogContentImage from "/src/components/BlogContentImage.astro";
+import BlogSummary from "/src/components/BlogSummary.astro";
+import islandDiag from "/src/content/blog/_images/future-of-astro-server-islands/dark-mode-server-islands-diag.webp";
+
+<BlogSummary>
+  Astro 4.12 is out now! This release includes a first look at [Server Islands](https://astro.build/blog/future-of-astro-server-islands/), as well as improvements to pagination and syntax highlighting, and more.
+</BlogSummary>
+
+This release includes the following highlights:
+
+- [**Experimental: Server Islands**](#experimental-server-islands)
+- [**Added `first` and `last` URLs to pagination data**](#added-first-and-last-urls-to-pagination-data)
+- [**Added support for the Shiki's `defaultColor` option**](#added-support-for-the-shikis-defaultcolor-option)
+- [**New `inferRemoteSize` function**](#new-inferremotesize-function)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Server Islands
+
+Today we're announcing experimental support for **Server Islands**, bringing islands architecture to the server. With Server Islands you mark components to run dynamically on the server during page load, allowing the rest of the page to be cached at the edge.
```

Comment:
"bringing islands architecture to the server" is a little unclear (I get what you're tring to say here).

Our two messages from last time were "personalization" and "performance", any way to keep hitting those here? Alternatively, a phrase that I liked was the idea that this was "our biggest fundamental change to Islands architecture since the creation of Astro" which is attention grabbing but not super informative. Up to you what to do here!

Reply:
> With Server Islands you mark components to run dynamically on the server during page load, allowing the rest of the page to be cached at the edge.

It's not 100% clear how this is different from the way Astro and other web frameworks exist today, to the casual reader. You do a great job at this on the server-islands demo page, can you keep this simple but make it clear why this is a big deal?